src/components/Join/index.jsx

```diff
@@ -1,117 +1,178 @@
-import React, { useState, useRef } from "react";
+import React, { useState, useEffect } from "react";
 import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+import useAxios from "../../hooks/useAxios";
+import useInput from "../../hooks/useInput";
+import "./index.css";
 
 function Join() {
-  const idInputRef = useRef();
-  const pwInputRef = useRef();
-
-  const [joinInput, setJoinInput] = useState({
-    userId: "",
-    pw: "",
-    pwCheck: "",
-    email: "",
-    nickname: "",
-  });
-  const [pwAlert, setPwAlert] = useState("");
-
-  const handlePwCheck = () => {
-    if (setJoinInput.pw < 1 || setJoinInput.pwCheck < 1) {
-      setPwAlert("비밀번호를 입력하세요");
-      // 비밀번호가 같다면 일치
-    } else if (setJoinInput.pw === setJoinInput.pwCheck) {
-      setPwAlert("일치");
-      // 비밀번호가 같지 않다면 불일치
-    } else {
-      setPwAlert("불일치");
+  let formIsValid = false;
+  const { sendRequest: sendFormRequest } = useAxios();
+  const [idCheck, setIdCheck] = useState("");
+  const navigate = useNavigate();
+
+  const {
+    value: inputEmail,
+    isValid: inputEmailIsValid,
+    hasError: emailInputHasError,
+    handleValueChange: handleEmailChange,
+    handleInputBlur: handleEmailBlur,
+    reset: resetEmailInput,
+  } = useInput((value) => value.includes("@" && "."));
+
+  const {
+    value: inputPw,
+    isValid: inputPwIsValid,
+    hasError: pwInputHasError,
+    handleValueChange: handlePwChange,
+    handleInputBlur: handlePwBlur,
+    reset: resetPwInput,
+  } = useInput((value) => value.trim() !== "");
+
+  const {
+    value: inputPwReEnter,
+    hasError: pwReEnterInputHasError,
+    handleValueChange: handlePwReEnterChange,
+    handleInputBlur: handlePwReEnterBlur,
+    reset: resetPwReEnterInput,
+  } = useInput((value) => value === inputPw);
+
+  const {
+    value: inputNickName,
+    isValid: inputNickNameIsValid,
+    hasError: nickNameInputHasError,
+    handleValueChange: handleNickNameChange,
+    handleInputBlur: handleNickNameBlur,
+    reset: resetNickNameInput,
+  } = useInput((value) => value.trim() !== "");
+
+  if (inputEmailIsValid && inputPwIsValid && inputNickNameIsValid) {
+    formIsValid = true;
+  }
```

Comment:
`formIsValid`값은 이후 재할당되지 않으니, 상단에서 let에서 선언하는 대신 여기서 const로 선언할 수도 있겠습니다.
```suggestion
  const formIsValid = inputEmailIsValid && inputPwIsValid && inputNickNameIsValid;
```

Reply:
수정했습니다 🙇‍♀️🙇‍♂️